src/org/exist/xmldb/RemoteXMLResource.java

```diff
@@ -96,12 +120,14 @@ public RemoteXMLResource(
             final int handle,
             final int pos,
             final XmldbURI docId,
-            final Optional<String> id)
+            final Optional<String> id,
+            final Optional<String> type)
```

Comment:
shouldn't Optional's in parameters of constuctors/methods be avoided?

Reply:
I have seen some advice that says that yes, and some that doesn't. I am not 100% convinced yet...